spec/mongoid/railties/console_sandbox_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'spec_helper'
+require 'mongoid/railties/console_sandbox'
+
+describe 'console_sandbox' do
+  describe '#check_if_transactions_might_be_available!' do
+    context 'cluster may support transactions' do
+      require_topology :replica_set, :sharded, :load_balanced
+
+      it 'does not raise' do
+        expect do
+          check_if_transactions_might_be_available!(Mongoid.default_client)
+        end.not_to raise_error
+      end
+    end
+
+    context 'cluster does not support transactions' do
+      require_topology :single
+
+      it 'raises an error' do
+        expect do
+          check_if_transactions_might_be_available!(Mongoid.default_client)
+        end.to raise_error(Mongoid::Errors::TransactionsNotSupported)
+      end
+    end
+  end
+
+  describe '#start_sandbox_transaction' do
+    require_transaction_support
+
+    before do
+      start_sandbox_transaction(Mongoid.default_client)
+    end
+
+    after do
+      Mongoid.send(:_session).abort_transaction
```

Comment:
This is outside the scope of your PR, but I'm curious: do you know why `_session` is private? It's a bit of a red flag for me any time I see tests (or other code) use `#send` to invoke a private method. We have `default_client`; is there value in a `default_session` method that could expose this?

(Like I said---this is outside the scope of this PR. I'm not expecting any action on this, just curious what you think.)

Reply:
I do not think the `_session` must be private; it is probably enough to annotate it with `@api private`.
What is important is that Mongoid users should not use it since it is an internal implementation detail.